test/token/ERC721/ERC721.behavior.js

```diff
@@ -283,26 +294,29 @@ function shouldBehaveLikeERC721 (
         describe('to a receiver contract returning unexpected value', function () {
           it('reverts', async function () {
             const invalidReceiver = await ERC721ReceiverMock.new('0x42', false);
-            await shouldFail.reverting(
-              this.token.safeTransferFrom(owner, invalidReceiver.address, tokenId, { from: owner })
+            await shouldFail.reverting.withMessage(
+              this.token.safeTransferFrom(owner, invalidReceiver.address, tokenId, { from: owner }),
+              'ERC721: transfer to non ERC721Receiver implementer'
             );
           });
         });
 
         describe('to a receiver contract that throws', function () {
           it('reverts', async function () {
             const invalidReceiver = await ERC721ReceiverMock.new(RECEIVER_MAGIC_VALUE, true);
-            await shouldFail.reverting(
-              this.token.safeTransferFrom(owner, invalidReceiver.address, tokenId, { from: owner })
+            await shouldFail.reverting.withMessage(
+              this.token.safeTransferFrom(owner, invalidReceiver.address, tokenId, { from: owner }),
+              'ERC721ReceiverMock: reverting'
             );
           });
         });
 
         describe('to a contract that does not implement the required function', function () {
           it('reverts', async function () {
             const invalidReceiver = this.token;
-            await shouldFail.reverting(
-              this.token.safeTransferFrom(owner, invalidReceiver.address, tokenId, { from: owner })
+            await shouldFail.reverting.withMessage(
+              this.token.safeTransferFrom(owner, invalidReceiver.address, tokenId, { from: owner }),
+              'VM Exception while processing transaction: revert'
```

Comment:
Let's just use `reverting` for these. @frangio WDYT about performing a low-level call in these cases (i.e. when calling a function of an unknown contract, assuming it will implement a certain interface), to provide a better error message? (`ERC721Receiver call faled`).

Reply:
Oh that's interesting. Let's open an issue to explore that afterwards?

Reply:
Created https://github.com/OpenZeppelin/openzeppelin-solidity/issues/1727.